### DISCUSSION TITLE
[Worksheet] Import classes from project into script

### DISCUSSION THREAD

**Discussion by elbaulp** [scalameta/metals#2771] [Q&A]
Hello, I have a worksheet file inside my project, and also a regular scala file with classes I would like to import, so like this:

```
src/main/scala/test.worksheet.sc
src/main/scala/myscalafile.scala
```
[The documentation states](https://scalameta.org/metals/docs/editors/vscode.html#getting-started-with-worksheets) that as long as the worksheet file is inside the project, you can import things from the project, but I am unable to. 

When trying to import a class from my scala file I get a import error:

```scala
import myscalafile._ // not found: value myscalafile
```

What's the correct way of importing project objects into the sheet?

**Comment by gabro** [comment]
How does your file look like? You don't need to import the file, you should be able to reference the objects and classes in your scala file directly

**Reply by elbaulp** [reply]
Thanks for your quick reply 

The scala file looks like this:

```
package my.package

case object MainApp extends App {
// ....
}

case class MyClass(....) // The one I want to import
```

**Reply by gabro** [reply]
Have you tried to import my.package.MyClass?

Auto-import should work as well in Scala 2 worksheets

**Reply by elbaulp** [reply]
Yes, I did before asking, I'm getting `object package is not a member of package my (lsp)`

**Reply by tgodzik** [reply]
You need to do import my.`package` I think. It's a reserved keyword, so I would overall avoid using it.

**Reply by elbaulp** [reply]
Will try, `my.package` was an example, the actual one I'm using in my code is `com.elbauldelprogramador`. I'll let you know if it worked when I can try it. 

Thanks!
